### DISCUSSION TITLE
Route not showing anything

### DISCUSSION THREAD

**Discussion by lucas-davi** [opcodesio/log-viewer#201] [Q&A] [answered]
Hello guys

After installing the log-viewer when accessing the route i get the following return

![image](https://user-images.githubusercontent.com/60628727/221415295-7403fa40-7d73-4b09-a8bd-f1cfb57628ff.png)

Any ideas how to workaround this?

**Comment by arukompas** [comment]
Hey @lucas-davi ,

Can you please share what the browser console outputs?

**Reply by lucas-davi** [reply]
of course

![image](https://user-images.githubusercontent.com/60628727/221455077-f66dbb9d-0bc9-41cc-8b35-0996e6b9496d.png)

Thanks for support.

**Reply by arukompas** [reply]
@lucas-davi 
1. does it work in Incognito mode, or another browser?
2. Does it work on a brand new Laravel project? 
3. What environment are you running on? Windows/mac/linux, Valet/xampp/Docker/etc?

**Reply by Godehardt2003** [reply]
> @lucas-davi
> 
>     1. does it work in Incognito mode, or another browser?
> 
>     2. Does it work on a brand new Laravel project?
> 
>     3. What environment are you running on? Windows/mac/linux, Valet/xampp/Docker/etc?

I think I got the same issue. Searching now for more than three days and can't find the problem.

The index.blade.php is rendered. 
![image](https://user-images.githubusercontent.com/78535292/221654418-0a4e864e-922e-443b-aea1-02940c59a420.png)
But exactly where the router-view tag should be "rendered" it is replaced by ""

I created a completely new laravel project and still the same issue. But if it works for other people it may be laragon. I don't know how vue.js works but maybe there is something wrong with some file loading or whatever. I have no clue.

As I said I'm on Windows/laragon.
FOR ME also not working in incognito mode.

I did the following steps (c:\laragon\www):
```
composer create-project --prefer-dist laravel/laravel test
cd .\test\
composer require opcodesio/log-viewer
php artisan log-viewer:publish
```

Blank page on http://localhost/logtest/test/public/log-viewer

**Reply by lucas-davi** [reply]
> @lucas-davi
> 
> 1. does it work in Incognito mode, or another browser?
> 2. Does it work on a brand new Laravel project?
> 3. What environment are you running on? Windows/mac/linux, Valet/xampp/Docker/etc?

Hey @arukompas 

1. Not working for me, same error
2. Not working for me, same error (tested on windows)
3. I tested  in Windows (xampp) and Linux

**Reply by arukompas** [reply]
It seems like the `app.js` file isn't loading for some reason, maybe because of the setup.

@lucas-davi @Godehardt2003 can you please also test [Laravel Horizon](https://laravel.com/docs/10.x/horizon) and whether the frontend opens up for you? I feel like vendor assets are not loading in your setups. Can you please check the Network tab in Chrome to see what happens to the `app.js` file?

**Reply by Godehardt2003** [reply]
> It seems like the `app.js` file isn't loading for some reason, maybe because of the setup.
> 
> @lucas-davi @Godehardt2003 can you please also test [Laravel Horizon](https://laravel.com/docs/10.x/horizon) and whether the frontend opens up for you? I feel like vendor assets are not loading in your setups. Can you please check the Network tab in Chrome to see what happens to the `app.js` file?

The app.js itself is loading
![image](https://user-images.githubusercontent.com/78535292/222432550-d288cf11-d783-4dce-ad42-d4192502062c.png)
response is the file:
![image](https://user-images.githubusercontent.com/78535292/222432598-8d99ebe8-2498-4ec0-87b9-ced400e8e747.png)

If I enter composer require laravel/horizon I get the following error:
```
 Problem 1
    - laravel/horizon v0.1.0 requires illuminate/contracts ~5.4 -> found illuminate/contracts[v5.4.0, ..., v5.8.36] but these were not loaded, likely because it conflicts with another require.
    - laravel/horizon[v0.2.0, v1.0.0, ..., v1.4.3, v2.0.0, ..., v2.2.2, v3.0.0, ..., v3.7.2, v5.4.0, ..., v5.14.2] require ext-pcntl * -> it is missing from your system. Install or enable PHP's pcntl extension.
    - laravel/horizon[v4.0.0, ..., v4.3.5] require php ^7.2 -> your php version (8.2.3) does not satisfy that requirement.
    - laravel/horizon[v5.0.0, ..., v5.3.0] require php ^7.3 -> your php version (8.2.3) does not satisfy that requirement.
    - Root composer.json requires laravel/horizon * -> satisfiable by laravel/horizon[v0.1.0, v0.2.0, v1.0.0, ..., v1.4.3, v2.0.0, ..., v2.2.2, v3.0.0, ..., v3.7.2, v4.0.0, ..., v4.3.5, v5.0.0, ..., v5.14.2].
```
I'm not that good in composer and have no clue what that is now. ext-pcntl isn't available on windows.

**Reply by Godehardt2003** [reply]
> > It seems like the `app.js` file isn't loading for some reason, maybe because of the setup.
> > @lucas-davi @Godehardt2003 can you please also test [Laravel Horizon](https://laravel.com/docs/10.x/horizon) and whether the frontend opens up for you? I feel like vendor assets are not loading in your setups. Can you please check the Network tab in Chrome to see what happens to the `app.js` file?
> 
> The app.js itself is loading ![image](https://user-images.githubusercontent.com/78535292/222432550-d288cf11-d783-4dce-ad42-d4192502062c.png) response is the file: ![image](https://user-images.githubusercontent.com/78535292/222432598-8d99ebe8-2498-4ec0-87b9-ced400e8e747.png)
> 
> If I enter composer require laravel/horizon I get the following error:
> 
> ```
>  Problem 1
>     - laravel/horizon v0.1.0 requires illuminate/contracts ~5.4 -> found illuminate/contracts[v5.4.0, ..., v5.8.36] but these were not loaded, likely because it conflicts with another require.
>     - laravel/horizon[v0.2.0, v1.0.0, ..., v1.4.3, v2.0.0, ..., v2.2.2, v3.0.0, ..., v3.7.2, v5.4.0, ..., v5.14.2] require ext-pcntl * -> it is missing from your system. Install or enable PHP's pcntl extension.
>     - laravel/horizon[v4.0.0, ..., v4.3.5] require php ^7.2 -> your php version (8.2.3) does not satisfy that requirement.
>     - laravel/horizon[v5.0.0, ..., v5.3.0] require php ^7.3 -> your php version (8.2.3) does not satisfy that requirement.
>     - Root composer.json requires laravel/horizon * -> satisfiable by laravel/horizon[v0.1.0, v0.2.0, v1.0.0, ..., v1.4.3, v2.0.0, ..., v2.2.2, v3.0.0, ..., v3.7.2, v4.0.0, ..., v4.3.5, v5.0.0, ..., v5.14.2].
> ```
> 
> I'm not that good in composer and have no clue what that is now. ext-pcntl isn't available on windows.

So I don't know why but php artisan serve is working directly and a way more faster than laragon so I simply use that... I have no clue why it is not working in laragon

**Reply by lucas-davi** [reply]
> It seems like the `app.js` file isn't loading for some reason, maybe because of the setup.
> 
> @lucas-davi @Godehardt2003 can you please also test [Laravel Horizon](https://laravel.com/docs/10.x/horizon) and whether the frontend opens up for you? I feel like vendor assets are not loading in your setups. Can you please check the Network tab in Chrome to see what happens to the `app.js` file?

@arukompas I installed Laravel Horizon and the frontend showed 

![image](https://user-images.githubusercontent.com/60628727/222811315-a174d524-64ad-40ec-bd29-29ea978bc83e.png)

- Network tab in Chrome after accessing log-viewer route

![image](https://user-images.githubusercontent.com/60628727/222811619-42b43fdf-c89d-4635-8208-624a9b7e3464.png)

**Reply by arukompas** [reply]
@lucas-davi thanks a lot! Does it also work for you if you run the server using `php artisan serve`?

Overall, I might have to find myself a windows environment or something, because it's hard to debug like this :/

**Comment by arukompas** [comment, marked as answer]
Hey @lucas-davi @Godehardt2003 
I've set up a Windows machine and was able to figure it out.

The problem happened when the URL in the browser (e.g. `localhost/test/public/log-viewer`) did not match the expected path (e.g. `localhost/log-viewer`) and that's because of how Laragon/Xampp are configured. Laragon offers a feature to "auto-create virtual hosts" (check settings) which then creates new test domains for each project, like `http://test.test` instead of `http://localhost/test/public`. Maybe Xampp does something similar too, not sure.

**Anyway**, a fix for these situations is now ready in **v2.1.1** - please update :) 

Hope this helps!

**Comment by lucas-davi** [comment]
Hello @arukompas 

After updating everything works perfectly for me :)

![image](https://user-images.githubusercontent.com/60628727/222916683-f64517ec-56d3-4344-89ed-54b37a78f72b.png)

thanks for the help!
